ADO Extension should support nested directories for outputDir

## Describe the bug
If users pass a nested directory (e.g. "$(DefaultDirectoryPath)\parent\child')" to **outputDir**, and the parent directory doesn't exist yet, the extension should be able to create the parent directory and not throw an exception. Currently, the extension expects to be sent single layer/non-nested throws an error when expected to write to a parent directory that does not already exist. 

<!-- A clear and concise description of what the bug is. -->

**To Reproduce**
Steps to reproduce the behavior:

1. Navigate to the yaml file
2. Go to inputs
3. Go to outputDir and specify '$(DefaultDirectoryPath)\parent\child'
4. Try to build and receive the error "##[error][Exception]ErrorWithCause: An error occurred while scanning website page: undefined

## CodePen repro example

<!--
Please provide a reproduction of the bug in a codepen, if possible. Here’s how

* Goto https://codepen.io/ for a starting codepen
* Create a simple example of the page that you have issues with and Export that codepen or give us a link to that codepen.
-->

## Expected behavior
The extension should create the listed parent directory if it does not already exist. 
<!-- A clear and concise description of what you expected to happen. -->

## Screenshots

<!-- If applicable, add screenshots to help explain your problem. -->

## Context (please complete the following information)

- OS Name & Version: <!--e.g. Windows 10.0.18363 Build 18363 / macOS Catalina 10.15(19A583)-->
- Azure DevOps Extension Version & Environment: <!--e.g. 1.140.1 Insider-->
- Browser Version: <!--e.g. Google Chrome 78.0.3904.108 (Official Build) (64-bit)-->
- Target Page: <!--e.g. https://www.bing.com-->

## Are you willing to submit a PR?
Yes
<!-- If asked, will you be willing to submit a PR to fix this bug? -->

## Did you search for similar existing issues?
Yes
<!-- Did you search for similar issues in our github issues or on [stackoverflow](https://stackoverflow.com/questions/tagged/accessibility-insights) -->

## Additional context
This issue was brought to our attention on 3/27 through the support alias
<!-- Add any other context about the problem here. -->

--- a/packages/shared/src/scanner/scanner.spec.ts
+++ b/packages/shared/src/scanner/scanner.spec.ts
@@ -405,7 +405,7 @@ describe(Scanner, () => {
                 .returns(() => false)
                 .verifiable();
             // eslint-disable-next-line security/detect-non-literal-fs-filename
-            fsMock.setup((fsm) => fsm.mkdirSync(reportOutDir)).verifiable();
+            fsMock.setup((fsm) => fsm.mkdirSync(reportOutDir, { recursive: true })).verifiable();
 
             const crawlerParams: CrawlerRunOptions = {
                 baseUrl: scanArguments.url,

--- a/packages/shared/src/scanner/scanner.ts
+++ b/packages/shared/src/scanner/scanner.ts
@@ -194,7 +194,7 @@ export class Scanner {
         if (!this.fileSystemObj.existsSync(outDirectory)) {
             this.logger.logInfo(`Report output directory does not exist. Creating directory ${outDirectory}`);
             // eslint-disable-next-line security/detect-non-literal-fs-filename
-            this.fileSystemObj.mkdirSync(outDirectory);
+            this.fileSystemObj.mkdirSync(outDirectory, { recursive: true });
         }
     }
 

